tests/test_eigen_tensor_no_array.cpp

```diff
@@ -0,0 +1,13 @@
+/*
+    tests/eigen_tensor.cpp -- automatic conversion of Eigen Tensor
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+constexpr const char *TEST_MODULE_NAME = "eigen_tensor_no_array";
+#define EIGEN_AVOID_STL_ARRAY
+
+#define TEST_NAMESPACE eigen_tensor_no_array
+
+#include "test_eigen_tensor.ipp"
```

Comment:
Could you please make this test_eigen_tensor.inl?

I know this works without trouble with our toolchain, I'm using .inl already on the smart_holder branch. (Back then I tried other extensions and not all worked; I forget exactly which worked and which don't.)